packages/react-native-reanimated/src/Colors.ts

```diff
@@ -164,6 +164,13 @@ function parsePercentage(str: string): number {
   return int / 100;
 }
 
+function clampRGBA(RGBA: ParsedColorArray): void {
+  'worklet';
+  for (let i = 0; i < 4; i++) {
+    RGBA[i] = Math.max(0, Math.min(RGBA[i], 1));
+  }
+}
+
```

Comment:
Why do we need this?

Reply:
when we used color based prop (i.ex. backgroundColor or boxShadow) with `withSpring` animation the colors would flicker, as the RGBA values were going below `0` and returning `NaN`. This function is a guard function that makes sure RGBA values stays within the given limit

Reply:
Let's make this bit a separate PR then to limit the scope of this one.